addons/event_crm_sale/models/sale_order.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _update_registrations(self, confirm=True, cancel_to_draft=False, registration_data=None):
+        registrations_before = self.env['event.registration'].search([])
+        res = super(SaleOrderLine, self)._update_registrations(confirm, cancel_to_draft, registration_data)
+        registrations_after = self.env['event.registration'].search([])
+        registrations = registrations_after - registrations_before
+        event_crm = self.env['event.crm'].search([])
+        if len(registrations) > 0:
+            if not registrations[0].web_registration:
```

Comment:
is it possible to mix registration that are web_registration with registration that are not ?

Reply:
not done yet